app/src/main/java/com/amazonaws/kinesisvideo/webrtc/URLVideoCapturer.java

```diff
@@ -0,0 +1,100 @@
+package com.amazonaws.kinesisvideo.webrtc;
+
+import android.content.Context;
+import android.net.Uri;
+
+import org.videolan.libvlc.LibVLC;
+import org.videolan.libvlc.Media;
+import org.videolan.libvlc.MediaPlayer;
+import org.videolan.libvlc.interfaces.IVLCVout;
+import org.webrtc.CapturerObserver;
+import org.webrtc.SurfaceTextureHelper;
+import org.webrtc.VideoCapturer;
+import org.webrtc.VideoFrame;
+import org.webrtc.VideoSink;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * This class implements a VideoCapturer for media streams. Similar to how camera capturers work, this
+ * class also leverages the SurfaceTextureHelper to do the heavy lifting of grabbing and encoding video
+ * frame using an intermediate OpenGL texture. It also uses libVLC(User must add) to playback streams on the texture.
+ */
+public class URLVideoCapturer implements VideoCapturer, VideoSink {
+    private String url;
+    private String[] options;
+    private String aspectRatio;
+    private SurfaceTextureHelper surfaceTextureHelper;
+    private Context context;
+    private CapturerObserver capturerObserver;
+
+    /**
+     * Public constructor that accepts the stream url.
+     * @param url Media stream url.
```

Comment:
/nit other two params and formatting

Reply:
Added JavaDoc comments for the whole file, and formatted. Kindly review.